src/corecel/math/Integrator.hh

```diff
@@ -0,0 +1,129 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/math/Integrator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+
+#include "corecel/Assert.hh"
+#include "corecel/Types.hh"
+
+#include "Algorithms.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+//! Solver options
+struct IntegratorOptions
+{
+    real_type epsilon{1e-3};  //!< Convergence criterion
+    int max_depth{30};  //!< Maximum number of outer iterations
```

Comment:
I reduced this from 1000 because each "trial" doubles the integer counter, so we'd overflow in many cases before reaching even 32.